### DISCUSSION TITLE
[Design] Recommended pattern for a simple QUIC Connection with a read and write stream

### DISCUSSION THREAD

**Discussion by ReeceHumphreys** [aiortc/aioquic#494] [Q&A]
Hello,

I want to establish a QUIC connection and then open a read and write stream that I can use to interface with this connection. I see on https://github.com/aiortc/aioquic/issues/169 that using the `create_stream` api is not recommended anymore. What would be the proper way to go about this with the event api? Im not looking to use Http/3 or anything like that. Just a simple QUIC connection.

Thanks!

**Comment by rthalley** [comment]
See [doq_client.py](https://github.com/aiortc/aioquic/blob/main/examples/doq_client.py) for an example that is suitable if you're only using one stream at a time.  If you have multiple concurrent streams, then your event dispatcher has to direct input to the right stream context.

I don't use the create_stream API in my DNS code that is built on top of aioquic because I either don't use asyncio or have to support asyncio as well as sync I/O and Trio.  I make stream abstractions that are individually blockable for reads on top of aioquic, so I don't think the concept is inherently bad.
